tests/lunamark/ext_mark.test

```diff
@@ -0,0 +1,5 @@
+lunamark -Xmark
+<<<
+Some ==highlighted text==.
+>>>
+<p>Some <span class="mark">highlighted text</span>.</p>
```

Comment:
Last time I checked, Pandoc just expands a span here, so I went for the simplest choice.
Of course we can possibly support outputting a `<mark>` in the HTML5 writer - either as suggested with a mark() function in writers, **or** possibly in the HTML5 writer itself, interpretating that span (avoiding yet another writer method for so small a thing).

Reply:
My distro didn't have a recent enough Pandoc for me to check, but [Pandoc tests][1] suggest that Pandoc also produces the `<mark>` element. Regardless, it seems clear to me that we should produce semantic HTML5 markup like `<mark>` whenever possible.

 [1]: https://github.com/jgm/pandoc/blob/edbea270f5ed0cae23cfe977b9621045256e7dfc/test/command/7743.md?plain=1#L1-L6

Reply:
It's not part of the Pandoc AST https://hackage.haskell.org/package/pandoc-types-1.23/docs/Text-Pandoc-Definition.html#t:Inline

And indeed `==mark==` processed with `-f markdown+mark -t json` gives

```
{"pandoc-api-version":[1,23,1],"meta":{},"blocks":[{"t":"Para","c":[{"t":"Span","c":[["",["mark"],[]],[{"t":"Str","c":"mark"}]]}]}]}
```

If I correctly read the code, the Pandoc Markdown reader indeed uses a span here:
https://github.com/jgm/pandoc/blob/edbea270f5ed0cae23cfe977b9621045256e7dfc/src/Text/Pandoc/Readers/Markdown.hs#L1726

Wondering where it does the mapping then! But it might thus likely be somewhere in Writers?

Reply:
Regardless, done in 96c05c8

Reply:
> Wondering where it does the mapping then! But it might thus likely be somewhere in Writers?

Seems that way. I was able to pinpoint a number of writers, which produce special output for spans with the `mark` class name:

- https://github.com/jgm/pandoc/blob/2c6f22d2b4edd9fb8b8576995d7aece99feef207/src/Text/Pandoc/Writers/LaTeX.hs#L765
- https://github.com/jgm/pandoc/blob/2c6f22d2b4edd9fb8b8576995d7aece99feef207/src/Text/Pandoc/Writers/Docx.hs#L1193
- https://github.com/jgm/pandoc/blob/2c6f22d2b4edd9fb8b8576995d7aece99feef207/src/Text/Pandoc/Writers/OpenDocument.hs#L611
- https://github.com/jgm/pandoc/blob/2c6f22d2b4edd9fb8b8576995d7aece99feef207/src/Text/Pandoc/Writers/RST.hs#L755
- https://github.com/jgm/pandoc/blob/2c6f22d2b4edd9fb8b8576995d7aece99feef207/src/Text/Pandoc/Writers/Markdown/Inline.hs#L351

I wasn't able to pinpoint where the HTML writer does it.